File: split/initterminal.pas
function initterminal: boolean;
label
  10;
begin
  reset(termin, 'TTY:', '/O/I');
  while true do
  begin
    ;
    write(termout, '**');
    break(termout);
    if not inputln(termin, true) then
    begin
      writeln(termout);
      write(termout, '! End of file on the terminal... why?');
      initterminal := false;
      goto 10;
    end;
    curinput.locfield := First;
    while (curinput.locfield < last) and (buffer[curinput.locfield] = 32) do
      curinput.locfield := curinput.locfield + 1;
    if curinput.locfield < last then
    begin
      initterminal := true;
      goto 10;
    end;
    writeln(termout, 'Please type the name of your input file.');
  end;
  10: ;
end;

File: cpp/src/initterminal.cpp
#include "initterminal.h"
#include "inputln.h"
#include <iostream>

void initterminal(void)
{
	while (true)
	{
		std::cout << "**" << std::flush;
		std::cin.clear();
		if (!inputln(std::cin, false))
		{
			std::cout << std::endl;
			std::cout << "! End of file on the terminal... why?'";
			throw std::string();
		}
		loc = First;
		while (loc < last && buffer[loc] == ' ')
			loc++;
		if (loc < last)
			return;
		std::cout << "Please type the name of your input file.\n";
	}
}
